playwright-e2e/lib/widget/dropdown.ts

```diff
@@ -6,21 +6,30 @@ import { Locator, Page } from '@playwright/test';
 export default class Dropdown {
   private readonly page: Page;
   private readonly locator: Locator;
+  private readonly label: string;
 
-  constructor(page: Page, label: string | RegExp) {
+  constructor(page: Page, label: string, opts: { exactMatch?: boolean } = {}) {
+    const { exactMatch = true } = opts;
     this.page = page;
-    this.locator = this.page.locator('li.dropdown').filter({ has: this.page.locator('a.dropdown-toggle'), hasText: label });
+    this.label = label;
+    const textSelector = exactMatch ? `text="${this.label}"` : `text=${this.label}`;
+    this.locator = this.page.locator('li.dropdown').filter({ has: this.page.locator(textSelector) });
   }
 
   toLocator(): Locator {
     return this.locator;
   }
 
+  getLabel(): string {
+    return this.label;
+  }
+
   /**
    * Determine if dropdown is open by look up aria-expanded property
    */
   async isOpen(): Promise<boolean> {
-    return (await this.toLocator().locator('a.dropdown-toggle').getAttribute('aria-expanded')) === 'true';
+    const ariaExpanded = await this.toLocator().locator('a[data-toggle="dropdown"]').getAttribute('aria-expanded');
+    return ariaExpanded ? Boolean(ariaExpanded) : false;
```

Comment:
Handles Null pointer error because `getAttribute('aria-expanded')` also returns `null` if attribute is not found.